### DISCUSSION TITLE
Remove multiple deployments

### DISCUSSION THREAD

**Discussion by sebalaini** [vercel/community#1117] [Help] [answered]
I saw an old answered question but I thought that I might ask the same as in 2 years things can change.

How can I delete all old production deployments?
I found this tutorial as well https://www.turfemon.com/delete-old-vercel-production-deployments but I didn't try it yet, any suggestions on how to achieve it?

The reason is just to remove all unused deployments and keep things clean and tidy :)

**Comment by amyegan** [comment]
Good question @sebalaini! The fastest way, if you're okay with deleting all but the single most recent production and preview deployments, is to use [`vercel remove [project-name] --safe`](https://vercel.com/docs/cli/remove#safe). If you need more control over what is removed, you have other CLI and API options.

You can use the API to [list deployments](https://vercel.com/docs/rest-api#endpoints/deployments/list-deployments) and loop through the list to [delete individual deployments](https://vercel.com/docs/rest-api#endpoints/deployments/delete-a-deployment). Or you can use [`vercel list`](https://vercel.com/docs/cli/list#) and [`vercel remove`](https://vercel.com/docs/cli/remove#) from the CLI to delete a list of deployments all at once.

**Reply by tresabhi** [reply]
This works but I'd still rather have good stale management

**Reply by sebalaini** [reply]
@tresabhi I agree with you that there should be an option to set it in Vercel itself and forget completely to cleanup deployment from time to time

**Reply by leerob** [reply]
We are working on a way to make it easier to clean up old deployments. Stay tuned!

**Reply by tresabhi** [reply]
Sounds promising, glad to hear an update about it!

**Reply by dnyg** [reply]
This is giving me an error constantly:

Error: Too many requests - try again in 5 minutes (more than 300, code: "now-alias-get").

**Reply by tresabhi** [reply]
@dnyg If you have a LOT of deployments, the Vercel CLI will reach the API call limit. So, wait for the amount of time mentioned and use the command again. This is normal.

**Comment by sebalaini** [comment]
Thanks, @amyegan that works perfectly for me :)

**Comment by leerob** [comment, marked as answer]
You can now configure the retention period for deployments through the dashboard and CLI.

For example, canceled and errored deployments might be set to 30 days retention, while production deployments might be set to 1 year. Recently deleted deployments are shown in your project settings and can be instantly restored within 30 days of deletion.

Learn more in our [documentation](https://vercel.com/docs/security/deployment-retention).

[a comment by an automated account is not shown]
